autogen/agentchat/conversable_agent.py

```diff
@@ -1793,13 +1809,13 @@ async def a_generate_reply(
         if messages is None:
             messages = self._oai_messages[sender]
 
-        # Call the hookable method that gives registered hooks a chance to process all messages.
+        # Call the hookable method that gives registered hooks a chance to process the last message.
         # Message modifications do not affect the incoming messages or self._oai_messages.
-        messages = self.process_all_messages_before_reply(messages)
+        messages = await self.a_process_last_received_message(messages)
```

Comment:
The ordering between the calling of `process_all_messages_before_reply` and `process_last_received_message` recently changed. Have you merged from main since then?